sos/report/plugins/ceph_osd.py

```diff
@@ -0,0 +1,100 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+import glob
+
+
+class CephOSD(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    short_desc = 'CEPH osd'
+
+    plugin_name = 'ceph_osd'
+    profiles = ('storage', 'virt', 'container')
+    containers = ('ceph-osd.*',)
+
+    def check_enabled(self):
+        return True if glob.glob('/var/lib/ceph/osd/*/*') else False
+
+    def setup(self):
+        self.add_file_tags({
+            '/var/log/ceph/ceph-osd.*.log': 'ceph_osd_log',
+        })
+
+        # Only collect OSD specific files
+        self.add_copy_spec([
+            "/var/log/ceph/ceph-osd*.log",
+            "/var/log/ceph/ceph-volume*.log",
+
+            "/var/lib/ceph/osd/",
+            "/var/lib/ceph/bootstrap-osd/",
+
+            "/run/ceph/ceph-osd*"
+        ])
+
+        self.add_cmd_output([
+            "ceph osd metadata",
+            "ceph osd erasure-code-profile ls",
+            "ceph osd crush show-tunables",
+            "ceph osd crush dump"
+        ])
+
+        ceph_cmds = [
+            "osd tree",
```

Comment:
I think most of these commands, if not all,  are actually meant to be run on the mon. You can test by running these commands directly on a osd node.

(With admin keys, you can anything anywhere - typically you'd expect to have it only on mon nodes).

Reply:
ack.  agreed, these would only run on a mon, i will move them all there. So then we have a few commands that run on OSDs like  ceph-disk list and ceph-volume inventory , ceph -v that runs in common.. they do not need a keyring passed to them. Shall I look for more examples, so we have at least one or two commands in each plugin for now? Or for the mgr/rgw/mds would you suggest keeping them totally empty stubs for now, with just some specific log files collected?

Reply:
ack